### ISSUE TITLE
Subtotals

### ISSUE THREAD

**Issue by korotkevics** [komarovalexander/ka-table#161]
Hi! Great work!
I'd like to ask you if the widget supports subtotals (summaries local to groupings)?
I'd like to be able to consolidate values as to my custom logic for each column within a group (e.g. sum values, deduct a single string value based on many etc.)
So when a group is collapsed, there is a local summary.
Is there a way to build it around an existing capability?

**Comment by komarovalexander** [comment]
Hi @korotkevics 
1) If you want to show some summary info in the group row, you can do it this way 
https://stackblitz.com/edit/table-grouping-custom-cell-ts-rmagyp?file=Demo.tsx (small note regarding demo: group icons are not shown in the stackblitz  :-) - but it should be fine in a real project)

2) if you want to show summary as an additional row at the end of grouping - this is not possible at this time - it is in progress and will be released by the end of this month 

Cheers!

**Comment by korotkevics** [comment]
> Hi @korotkevics
> 
> 1. If you want to show some summary info in the group row, you can do it this way
>    https://stackblitz.com/edit/table-grouping-custom-cell-ts-rmagyp?file=Demo.tsx (small note regarding demo: group icons are not shown in the stackblitz  :-) - but it should be fine in a real project)
> 2. if you want to show summary as an additional row at the end of grouping - this is not possible at this time - this is in progress and will be released by the end of this month
> 
> Cheers!

Thanks for the quick response.
I'd like having something like this: https://komarovalexander.github.io/ka-table/#/grouping-custom-row
But ,
- with groups collapsed by default (can be achieved currently with `groupsExpanded: {}`)
- with values consolidated for each column with a custom function that are displayed as normal cells (under an appropriate column) when a group is collapsed. So practically a group line becomes a normal row with consolidated values when collapsed.

I am not sure if you have anything like this in plans ;)
For now, I am consolidating values in advance and then rendering them as normal rows, but of course then there is no way to see details.

**Comment by komarovalexander** [comment]
Seems like you need TreeMode - then is not supported by ka-table.
but
1) ka-table also has detailsRow: 
https://komarovalexander.github.io/ka-table/#/details-row
probably it could be useful for you 
 or
2) you can render columns inside custom groupRow

**Comment by korotkevics** [comment]
> Seems like you need TreeMode - then is not supported by ka-table.
> but
> 
> 1. ka-table also has detailsRow:
>    https://komarovalexander.github.io/ka-table/#/details-row
>    probably it could be useful for you
>    or
> 2. you can render columns inside custom groupRow

Thanks!

1. I'm  now trying exactly this. I am keeping both plain and consolidated rows in a state, and I am resolving plain records based on a key when expanding details. I doubt the performance will be good, we will see. The CSS was a little bit stubborn, had to create a custom table inside.
2. Tried this in the first place, couldn't get around the native CSS.

**Comment by komarovalexander** [comment]
> 2. Tried this in the first place, couldn't get around the native CSS.

maybe you could give me an example I could have a look

**Comment by korotkevics** [comment]
> > 1. Tried this in the first place, couldn't get around the native CSS.
> 
> maybe you could give me an example I could have a look

I have already erased it, unfortunately.
Essentially, the style used by the group row was not giving in to be tweaked and turned into a regular row. Maybe it is possible, I am new to this thing.
I have managed to make it work with the detailed row: I am creating a child table on the expansion, and it is set to take 100% of width, and its column proportions are set to match header column proportions. I am keeping two Redux states: one with plain rows, and one with consolidated rows (being rendered in the first places). When generating rows in the details child table (on the expansion as I said earlier) I am retrieving plain rows with a shared key from a consolidated row.
